crates/pallet-executor/src/lib.rs

```diff
@@ -308,6 +324,29 @@ mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
+    fn check_execution_receipt(
+        SignedExecutionReceipt {
+            execution_receipt,
+            signature,
+            signer,
+        }: &SignedExecutionReceipt<T::SecondaryHash>,
+    ) -> Result<(), Error<T>> {
+        let msg = execution_receipt.hash().encode();
```

Comment:
do we need to encode the hash again ?
can we use the hash or encoded ex reciept as the payload

Reply:
Honstly I'm not 100% this is the final sign payload we'll use, but indeed an optimization!